src/lib.rs

```diff
@@ -1093,6 +1113,17 @@ impl<K, V, S> Drop for FlurryHashMap<K, V, S> {
     }
 }
 
+impl<K, V, S> FromIterator<(K, V)> for FlurryHashMap<K, V, S> {
+    fn from_iter<T: IntoIterator<Item = (K, V)>>(iter: T) {
+        // safety: we have &mut self, so not concurrently accessed by anyone else
+        let guard = unsafe { crossbeam::epoch::unprotected() };
+        let output = Self::new();
```

Comment:
I think I would use `Self::with_capacity` here in association with [`Iterator::size_hint`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint).

Reply:
Makes sense. Sinse the `FlurryHashMap::extend` method does not preallocate space for the incoming entries, this should be more efficient.

By the way, couldn't we use a similar strategy for the `extend` method?

Reply:
Oh, yes, you're right, we don't need `ExactSizeIterator` and specialization since `size_hint` is available on all `Iterator` types! You would need to port the Java `tryPresize` method to use this in `extend` though, since you are extending an _existing_ map and therefore cannot use `with_capacity`.

Reply:
Alright, I'll update the TODOs then.